src/openforms/submissions/tests/form_logic/test_binding.py

```diff
@@ -0,0 +1,375 @@
+from functools import lru_cache
+from pathlib import Path
+from typing import Any
+from unittest import skip
+from urllib.parse import unquote
+
+from django.test import SimpleTestCase
+
+import requests_mock
+from factory.django import FileField
+from furl import furl
+from hypothesis import example, given, settings, strategies as st
+from zgw_consumers.constants import APITypes, AuthTypes
+
+from openforms.forms.tests.factories import FormVariableFactory
+from openforms.registrations.contrib.zgw_apis.tests.factories import ServiceFactory
+from openforms.variables.constants import DataMappingTypes
+from openforms.variables.tests.factories import ServiceFetchConfigurationFactory
+from openforms.variables.validators import HeaderValidator, ValidationError
+
+from ...form_logic import bind
+
+DEFAULT_REQUEST_HEADERS = {
+    "Accept",
+    "Accept-Encoding",
+    "Connection",
+    "Content-Type",
+    "User-Agent",
+}
+
+
+def data_mapping_values() -> st.SearchStrategy[Any]:
+    "Generates values for the value side of openforms.typing.DataMapping"
+    return st.one_of(st.text(), st.integers(), st.floats(), st.dates(), st.datetimes())
+
+
+# the first test still takes too much time
+# decorating given because I don't want to count on test running order
+given = settings(deadline=500)(given)
+
+
+class ServiceFetchConfigVariableBindingTests(SimpleTestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.service = ServiceFactory.build(
+            api_type=APITypes.orc,
+            api_root="https://httpbin.org/",
+            auth_type=AuthTypes.no_auth,
+            oas_file=FileField(
+                from_path=str(Path(__file__).parent.parent / "files" / "openapi.yaml")
+            ),
+        )
+        # prevent parsing the yaml over and over and over
+        cls.service.id = 1  # need pk for __hash__
+        cls.service.build_client = lru_cache(1)(cls.service.build_client)
```

Comment:
this should be cached in memory by the client though :thinking: See https://github.com/maykinmedia/zgw-consumers/blob/main/zgw_consumers/cache.py#L11

Reply:
Yeah, that's what I thought. I've stepped through that line more than I'd like to remember.

Reply:
perhaps some dummy cache is being setup instead? https://github.com/open-formulieren/open-forms/blob/master/src/openforms/conf/ci.py#L14

Reply:
![image](https://user-images.githubusercontent.com/325643/219092583-921542f8-f115-43ae-812a-79c8dfcf1cda.png)